spring-data-relational/src/main/java/org/springframework/data/relational/core/dialect/OrderByOptionsSupport.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.relational.core.dialect;
+
+import org.springframework.data.domain.Sort;
+import org.springframework.lang.Nullable;
+
+/**
+ * The {@code ORDER BY} options supported by a specific Dialect.
+ *
+ * @author Chirag Tailor
+ */
+public interface OrderByOptionsSupport {
+	String resolve(@Nullable Sort.Direction direction, Sort.NullHandling nullHandling);
```

Comment:
I also think this function lacks documentation. It is not clear from the first glance, ( at leas for me :) )  what exactly resolve function does. Maybe we can add javadoc upon it? @ctailor2 Chirag, what do you think?

Reply:
Agreed @Mikhail2048, it wasn't very clear and I think that is largely because the two related but distinct concepts of `Sort.Direction` and `Sort.NullHandling` shouldn't be coupled together here. I moved the direction handling back out and and repurposed this interface to be for `Sort.NullHandling` only which hopefully makes the code more clear. I also added some more javadocs. Could you review my latest changes to see if these updates improve upon this?